### PR TITLE
[RF] Improvements to `RooAbsPdf::paramOn()` formatting

### PR DESCRIPTION
**Pull request by guitargeek**
Two changes: 

1. Removes the deprecated `RooFit::Format(const char* option, int)` command argument
2. Remove deprecated `RooAbsPdf::paramOn()` overload that take a formatting string directly
3. Make it possible to use parameter titles instead of variables in `RooAbsPdf::paramOn()`, closing the followng JIRA ticket:
    https://sft.its.cern.ch/jira/browse/ROOT-6039

More detail in the commit descriptions.

### CODE REVIEW

File: roofit/roofitcore/src/RooAbsPdf.cxx [hunk 1]
```diff
@@ -3190,13 +3152,10 @@ RooPlot* RooAbsPdf::paramOn(RooPlot* frame, const RooAbsData* data, const char *
 /// values specify the initial relative position of the text box in the plot frame.
 
 RooPlot* RooAbsPdf::paramOn(RooPlot* frame, const RooArgSet& params, bool showConstants, const char *label,
```
Comment (lmoneta) [inline roofit/roofitcore/src/RooAbsPdf.cxx:3154]: Here you are changing the `paramOn` interface. Can the old one be kept for backward compatibility and mark as deprecated and maybe removed later on ?

**Review by lmoneta — APPROVED** [review]
LGTM!
Thanks for fixing this old JIRA issue.
I have just a comment the changes in the `paramOn` interface

File: roofit/roofitcore/src/RooAbsPdf.cxx (hunk 1 above)
Comment (guitargeek) [reply roofit/roofitcore/src/RooAbsPdf.cxx:3154]: This is `private` function that is an implementation detail which is called by the public `paramOn()` function. So it doesn't matter if the change the interface

Comment (lmoneta) [reply roofit/roofitcore/src/RooAbsPdf.cxx:3154]: Good! I missed it was a private function.

**Pull request merged**